core/src/main/java/org/elasticsearch/index/query/BaseTermQueryBuilder.java

```diff
@@ -146,18 +146,18 @@ public QueryValidationException validate() {
     }
 
     @Override
-    protected final int doHashCode() {
-        return Objects.hash(getClass(), fieldName, value);
+    protected int doHashCode() {
+        return Objects.hash(fieldName, value);
```

Comment:
oh wait we already take care of that in the base class, forgot about that, I think it's fine then